### DISCUSSION TITLE
Issue with having key 'modelname'+'Id' created as an integer by user

### DISCUSSION THREAD

**Discussion by Sreenuraj** [radical-sh/public-docs#12] [Issues]
[codeError.zip](https://github.com/radical-sh/public-docs/files/14401503/codeError.zip)

- For a project I am able to generate code in tech stack: "Node - Experimental" but not in "Java"
- Since am unable to access the logs, attaching the recording and the URL: https://radical.sh/organizations/1d32d517-6684-5dbe-91ea-ed533bf3da17/namespaces/08756d72-018f-5b3f-bb6b-1529b654fb15/projects/f617470a-0ded-45a8-a191-5f95bb7f0636/designer/model

**Comment by Sreenuraj** [comment]
Looks like currently we can make only property whose datatype is string as keys. 
But some how I was able to make an integer property as Key. Maybe that is the reason of failure.
![image](https://github.com/radical-sh/public-docs/assets/11501324/ec307914-f63c-462d-91e4-f89614c47538)

**Comment by Sreenuraj** [comment]
So found the problem. 

- if user define 'modelname'+'Id' as an integer when creating schema property, user will not have option to make this property as key.
- But in API user can make the same property as key. Because that is the default one.
Only for Java its failing and not from node.

**Comment by Sreenuraj** [comment]
@senthilsivanath  have a look

**Comment by senthilsivanath** [comment]
Resolved, radical allows string keys. 

Root cause:
If we have property as string but key auto creation from API tab make this as integer key.

**Comment by Sreenuraj** [comment]
Another way to make property with datatype as integer to be a key:

https://github.com/radical-sh/public-docs/assets/11501324/124d7470-33f9-4526-9e98-d1989e99cab8
